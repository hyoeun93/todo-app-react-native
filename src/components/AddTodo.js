import React, { Component } from 'react'
import { TextInput, TouchableOpacity, StyleSheet} from 'react-native'
import { View, Body, CheckBox } from 'native-base';
import { Ionicons } from '@expo/vector-icons';

export default class AddTodo extends Component {
    constructor(props) {
        super(props);

        this.state = {
            title: '',
            done: null
        }
    }

    onSubmit = () => {
        if(this.state.title.length > 0) this.props.onAdd(this.state.title);
        return null;
    }

    setStateUtil = (property, value = undefined) => {
        this.setState({
            [property]: value
        })
    }

    render() {
        return (
            <View style={styles.container}>
                <Body style={styles.bodystyle}>
                    <TextInput 
                        style={{ width: '90%'}}
                        placeholder="What needs to be done?"
                        autoFocus 
                        underLineColorAndroid="transparent"
                        underlineColor="transparent"
                        blurOnSubmit
                        onSubmitEditing={this.onSubmit}
                        onChangeText={changedTitle => this.setStateUtil('title', changedTitle)}
                        value={this.state.title}
                        onBlur={this.props.onBlur}
                    />
                </Body>
                <TouchableOpacity
                    onPress={() => this.props.onCancelDelete}
                    style={{ paddingLeft: 25, paddingRight: 15 }}
                >                  
                    <Ionicons
                        name="ios-trash"
                        color={`${this.state.title.length > 0 ? 'black' : 'grey'}`}
                        size={23}
                    />
                </TouchableOpacity>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: '100%',
        flexDirection: 'row',
        alignItems: 'center',
        paddingRight: 10,
        paddingTop: 5,
    },
    center: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    bodystyle: {
        flex: 1,
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
        paddingLeft: 25,
    }
  });