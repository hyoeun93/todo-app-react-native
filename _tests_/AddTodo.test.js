import React from 'react';
import renderer from 'react-test-renderer';
import AddTodo from '../src/components/AddTodoButton';

it('TodoItem snapShot', () => {
    const snap = renderer.create(
        <AddTodo />
    ).toJSON();
    expect(snap).toMatchSnapshot();
})