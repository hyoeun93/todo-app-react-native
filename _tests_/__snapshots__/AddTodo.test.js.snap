// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 20,
      "flexDirection": "column",
      "height": 56,
      "left": undefined,
      "position": "absolute",
      "right": 20,
      "top": undefined,
      "width": 56,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5067FF",
        "borderRadius": 28,
        "bottom": 0,
        "elevation": 4,
        "height": 56,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 2,
        "width": 56,
      }
    }
  >
    <Text />
  </View>
</View>
`;
